src/main/java/com/algorand/algosdk/crypto/LogicsigSignature.java

```diff
@@ -48,14 +48,10 @@ public LogicsigSignature(
     ) {
         this.logic = Objects.requireNonNull(logic, "program must not be null");
         this.args = args;
-        boolean verified = false;
-        try {
-            verified = Logic.checkProgram(this.logic, this.args);
-        } catch (IOException ex) {
-            throw new IllegalArgumentException("invalid program", ex);
-        }
 
-        assert verified;
+        // NOTE: Logic.checkProgram has been deprecated, use Logic.sanityCheckProgram instead
+        //       we no longer read from langspec.json, we do not need to catch IOException error
+        Logic.sanityCheckProgram(this.logic);
```

Comment:
@ahangsu Minor and optional:  For replacements _not_ exposed publicly, I don't think an inline citation is required.  If you prefer it, feel welcomed to keep as is.


```suggestion
        Logic.sanityCheckProgram(this.logic);
```